### ISSUE TITLE
Turn off auto-indexing for certain operations

### ISSUE THREAD

**Issue by dbinetti** [algolia/algoliasearch-django#251]
Hi, is it possible to turn off auto-indexing temporarily?  (Like signal disabling?)

I run a nightly operation that runs a denormalization that touches every row, but does not affect fields that are reflected in my search index.  However, since I'm touching every row it counts as an operation, and I'm quickly using up my allotment for the month.  is there a way to turn off indexing immediately prior to, and re-enable following, my operation?

Thanks.

**Comment by PLNech** [comment]
Hi @dbinetti, thanks for suggesting this! I definitely see how it could be useful.
We just need to consider how we could implement this thoughtfully, I'll tag it as an `enhancement` and we'll add it to the roadmap for this integration :)

**Comment by PLNech** [comment]
@julienbourdeau: if that works for you, I propose you give a try to implementing this. 

- Check out how we implemented auto-indexing via [Signals](https://docs.djangoproject.com/en/2.0/topics/signals/), e.g. [`post_save`](https://docs.djangoproject.com/en/2.0/ref/signals/#post-save).
- Propose a design that would let a user conditionally disable it
- If you're confident in the design, you can PR an implementation directly

I'll happily review your design/PR :slightly_smiling_face:

**Comment by dbinetti** [comment]
So perhaps this is a dumb question, but shouldn't I be able to do this myself?  I know the process to disable signals, but for whatever reason I can't load the receiver `def __post_save_receiver(self, instance, **kwargs):` in `registration.py`.  Is there some reason for this?

**Comment by clemfromspace** [comment]
Hello @dbinetti 

I managed to disable the signal by doing the following:

```python
post_save.disconnect(algolia_engine._AlgoliaEngine__post_save_receiver, sender=MyModel)
```

Note the strange `_AlgoliaEngine__post_save_receiver` name of the method, coming from the [
Name mangling](https://docs.python.org/3/tutorial/classes.html#private-variables)

Let me know if that help :)

**Comment by clemfromspace** [comment]
Hi again @dbinetti,

I opened #266, I think it should solve the issue your are having, let me know what you think :)

**Comment by dbinetti** [comment]
Yep, perfect @clemfromspace.  This is *exactly* what I had in mind!

**Comment by clemfromspace** [comment]
@dbinetti Hey there, we just released the version 1.7.0 containing the context decorator :)
I am working on updating the documentation right now, but you can already upgrade and test it like below:

```python
from algoliasearch_django.decorators import disable_auto_indexing

@disable_auto_indexing
def my_method():
    pass

with disable_auto_indexing(model=MyModel):
    pass
```

**Comment by dbinetti** [comment]
Yep, works perfectly. thanks!
